prow/tide/tide.go

```diff
@@ -516,34 +551,39 @@ func toSimpleState(s kube.ProwJobState) simpleState {
 
 // isPassingTests returns whether or not all contexts set on the PR except for
 // the tide pool context are passing.
-func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest) bool {
+func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest, bp branchProtection) bool {
 	log = log.WithFields(pr.logFields())
 	contexts, err := headContexts(log, ghc, &pr)
 	if err != nil {
 		log.WithError(err).Error("Getting head commit status contexts.")
 		// If we can't get the status of the commit, assume that it is failing.
 		return false
 	}
-	return len(unsuccessfulContexts(contexts)) == 0
+	return len(unsuccessfulContexts(contexts, bp)) == 0
 }
 
 // unsuccessfulContexts determines which contexts from the list are failed that
 // we care about. For instance, we do not care about our own context.
-func unsuccessfulContexts(contexts []Context) []Context {
+func unsuccessfulContexts(contexts []Context, bp branchProtection) []Context {
 	var failed []Context
+	combinedSet := sets.NewString()
 	for _, ctx := range contexts {
+		combinedSet.Insert(string(ctx.Context))
```

Comment:
Let's break this out -- the doc on this method reads `"unsuccessfulContexts determines which contexts from the list are failed"` --  contexts that are not in the list should not be returned? Maybe a method `missingContexts` -- but unsure of the usefulness of this actually as once you mark a context required, GitHub automatically adds it to every commit, so we're coding against GitHub acting up?

Reply:
Or actually on second thought, you are changing the semantics of the check entirely -- rename this method to reflect the fact that instead of identifying failed contexts from an owning set, you are checking for successful contexts from a required set.

Reply:
I don't think we are changing the semantic. Here, we still skip the test we don't care about, ie  tests that are not required, and we add required tests that are missing. (In theory this should not happen, as github will add them as pending by default, but it is a safe guard). I updated the documentation.